.github/workflows/validate-release.yml

```diff
@@ -0,0 +1,94 @@
+name: Validate Release
+
+on:
+  workflow_call:
+    outputs:
+      RELEASE_VALIDATION_RESULT:
+        description: The release validation result
+        value: ${{ jobs.validate-release.outputs.RELEASE_VALIDATION_RESULT }}
+
+jobs:
+  validate-release:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    outputs:
+      RELEASE_VALIDATION_RESULT: ${{ steps.validate-release.outputs.RELEASE_VALIDATION_RESULT }}
+      RELEASE_VALIDATION_MESSAGE: ${{ steps.validate-release.outputs.RELEASE_VALIDATION_MESSAGE }}
+      EXISTING_COMMENT_ID: ${{ steps.check-bot-comments.outputs.EXISTING_COMMENT_ID }}
+    steps:
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
+        with:
+          is-high-risk-environment: false
+          # Fetch everything because if this is a pull request we need to
+          # compare with the base branch
+          fetch-depth: 0
+      - name: Check for existing bot comments
+        if: ${{ github.event_name == 'pull_request' }}
+        id: check-bot-comments
+        run: |
+          set -euo pipefail
+          EXISTING_COMMENT_ID=$(gh api "/repos/${GITHUB_REPOSITORY}/issues/${GITHUB_EVENT_PULL_REQUEST_NUMBER}/comments" --jq '[.[] | select(.user.login == "github-actions[bot]") | select(.body | contains("<!-- METAMASKBOT-RELEASE-VALIDATION -->"))] | sort_by(.created_at) | .[-1].id')
+          if [[ -n "$EXISTING_COMMENT_ID" && "$EXISTING_COMMENT_ID" != "null" ]]; then
+            echo "EXISTING_COMMENT_ID=$EXISTING_COMMENT_ID" >> "$GITHUB_OUTPUT"
+          else
+            echo "EXISTING_COMMENT_ID=" >> "$GITHUB_OUTPUT"
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_REPOSITORY: ${{ github.repository }}
+          GITHUB_EVENT_PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
```

Comment:
Maybe consider using `actions/github-script` for the logic checking for existing comments, and commenting? We do something similar for the Snaps Directory, and it's a lot more readable than bash:

https://github.com/MetaMask/snaps-directory/blob/7c32fcede2fee59db3a0babb2b4a4de5d686ba58/.github/workflows/publish-staging.yml#L27-L58